web/modules/custom/weather_block/src/Plugin/Block/WeatherBlock.php

```diff
@@ -11,40 +19,144 @@
   id: "weather_block",
   admin_label: new TranslatableMarkup("Weather block"),
 )]
-class WeatherBlock extends BlockBase {
+class WeatherBlock extends BlockBase implements ContainerFactoryPluginInterface {
+  /**
+   * The Config Factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected ConfigFactoryInterface $configFactory;
+  /**
+   * The Logger service.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected LoggerChannelFactoryInterface $logger;
+
+  /**
+   * Constructor for WeatherBlock.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $configFactory, LoggerChannelFactoryInterface $logger) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $configFactory;
+    $this->logger = $logger;
```

Comment:
Better to use constructor promotion https://www.php.net/manual/en/language.oop5.decon.php#language.oop5.decon.constructor.promotion
It is just simpler